index.html

```diff
@@ -857,6 +852,91 @@
+\begin{equation} \label{interfaced_lambertian_albedo}
+E_c(\omega_o) = F(\omega_o, \eta_c) + E_b \Bigl(1 - F(\omega_o, \eta_c)\Bigr) \Delta(E_b, \eta_c)
+\end{equation}
+where $\Delta(E_b, \eta_c)$ is a darkening factor given by
+\begin{equation} \label{general_darkening_formula}
+\Delta(E_b, \eta_c) = \frac{1 - K}{1 - E_b K} \ .
+\end{equation}
+We can thus approximate the required boost factor as the reciprocal of this, i.e. $B_0 \approx \Delta^{-1}$. Note that as $E_b \rightarrow 1$, the darkening factor $\Delta \rightarrow 1$, as required since physically no energy is lost if the base reflects all incident energy (in the clear-coat case). If the base albedo $E_b$ is boosted by $B(\delta)$, then clearly at $\delta=0$ the darkening factor cancels and equation [undarkened_coat_albedo] is satisfied as required.
+
+Here $K \in [0,1]$ is the _internal diffuse reflection coefficient_, which corresponds to the fraction of the energy leaving the base which returns
+to the base due to internal reflection from the coat. Higher values of $K$ lead to more internal reflections, thus more darkening and saturation. In the case of a Lambertian base (which should be a reasonable approximation to the rough metal, dielectric, or diffuse cases), $K = K_r$ where, with relative coat IOR $\eta_c$:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_rough_base}
+K_r = 1 - \Bigl(1 - E_F(\eta_c)\Bigr)/\eta_c^2 \ .
+\end{equation}
+The albedo $E_F(\eta_c)$ is the _hemispherical_ (or _average_) albedo of the coat Fresnel factor [^avg_fresnel]. If the base has specular reflection (due to a smooth metallic or dielectric interface) then equation [general_darkening_formula] still applies, but the appropriate value of the internal diffuse reflection coefficient $K$ should be closer to:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_smooth_base}
+K_s = F(\omega_o, \eta_c) \ ,
+\end{equation}
+which becomes exact in the limit of a perfectly smooth metallic or dielectric base. There is significantly less darkening in the case of a smooth base, as the specular reflection from the base is less likely to generate total internal reflection at the coat boundary.
+
+We thus recommend in the general case that the darkening factor of equation [general_darkening_formula] be applied, but taking
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_general_base}
+K = \mathrm{lerp}(K_s, K_r, r_b)
+\end{equation}
+where $r_b$ is an estimate of the base roughness, a blend between metal $r_m$ and dielectric $r_d$ roughness estimates according to the **`base_metalness`** $\mathtt{M}$:
```

Comment:
> a blend between metal $r_m$ and dielectric $r_d$ roughness estimates

Readability might be improved by swapping the order of $r_m$ and $r_d$ to match the following formula:

> $r_b = \mathrm{lerp}(r_d, r_m, \mathtt{M})$

Reply:
> > a blend between metal rm and dielectric rd roughness estimates
> 
> Readability might be improved by swapping the order of rm and rd to match the following formula:
> 
> > rb=lerp(rd,rm,M)

Done in e3cb364f95bd795afe3c61a08835f30ee3e8dedb

Reply:
> > base roughness
> 
> Since `base_roughness` is the name of a parameter right now, it might be useful to use a different term than "base roughness" here (e.g., "effective base roughness" or something other than "base").

Done in a7f22fcddd71e944b5424df7a6c3a8618b303f71